tools/ci_build/build.py

```diff
@@ -684,10 +702,13 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
             "ON" if args.enable_nvtx_profile else "OFF"),
         "-Donnxruntime_ENABLE_TRAINING=" + (
             "ON" if args.enable_training else "OFF"),
-        "-Donnxruntime_USE_HOROVOD=" + (
-            "ON" if args.use_horovod else "OFF"),
+        # Enable advanced computations such as AVX for some traininig related ops.
+        "-Donnxruntime_ENABLE_ADVANCED_INSTRUCTIONS_FOR_TRAINING=" + (
+            "ON" if args.enable_training and check_all_flags_supported(['-mf16c', '-mavx', '-mfma']) else "OFF"),
```

Comment:
Does cmake handle these kinds of tests in a more general way (i.e., not running gcc directly, so it would work with clang, or pick up the appropriate settings for cross-compiling)?

Reply:
Good point on the cross-compilation, I will move this logic to cmakeList then.